app/queries/get_application_choices_for_providers.rb

```diff
@@ -21,5 +21,6 @@ def self.call(providers:, vendor_api: false, includes: DEFAULT_INCLUDES)
       .or(ApplicationChoice.includes(*includes)
         .where('courses.accredited_provider_id' => providers, 'courses.recruitment_cycle_year' => RecruitmentCycle.years_visible_to_providers))
       .where('status IN (?)', statuses)
+      .select('"application_choices"."id" AS id')
```

Comment:
This is an intersting one. The raw sql generated by `GetApplicationChoicesForProviders` labels the `id` column as `t0_r0`, probably because of the complexity of the query. This addition recasts `t0_r0` as `id`, which allows the resulting sql to be used in a join in a way consistent with any other `ApplicationChoice` Arel.

Because of this rename we can join on `id` as usual in the new service, which means you can write things like: `GetActivityLogEvents.new(application_choices: ApplicationChoice.where(....)).call` and expect them to work.

Reply:
This comment is for `GetActivityLogEvents`, not `GetApplicationChoicesForProviders`. Will post a reply on that file.